dts/bindings/gpio/nxp,pca_series.yaml

```diff
@@ -0,0 +1,55 @@
+# Copyright (c) 2023 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+description: PCA95xx series I2C-based GPIO expander
+
+compatible: "nxp,pca-series"
+
+include: [gpio-controller.yaml, i2c-device.yaml]
+
+properties:
+  part_no:
```

Comment:
I don't think this how we should support multiple devices with one driver. Zephyr drivers that support multiple compatibles typically do so via a pattern like so:

```
/* In this case, type would be a parameter indicating the part number of the device, based on compatible */
#define MY_DRV_INIT_MACRO(inst, type)

#define DT_DRV_COMPAT xxxx
DT_INST_FOREACH_STATUS_OKAY_VARGS(MY_DRV_INIT_MACRO, XXXX)

#undef DT_DRV_COMPAT
#define DT_DRV_COMPAT yyyy
DT_INST_FOREACH_STATUS_OKAY_VARGS(MY_DRV_INIT_MACRO, YYYY)
````

See the lp50xxx driver for an example of this method: https://github.com/zephyrproject-rtos/zephyr/blob/4c4ff203837bd17ffbcd5e5dc8bd90818dda74b4/drivers/led/lp50xx.c#L400

Another approach we could use would be something like what the ILI9XXX driver does: https://github.com/zephyrproject-rtos/zephyr/blob/4c4ff203837bd17ffbcd5e5dc8bd90818dda74b4/drivers/display/display_ili9xxx.c#L411

I think either approach would be fine- but we should avoid encoding part number like this

Reply:
use "DT_INST_FOREACH_STATUS_OKAY_VARGS" to pass part_no.